extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSearchExtractor.java

```diff
@@ -133,15 +353,217 @@ private void collectStreamsFrom(InfoItemsSearchCollector collector, JsonArray vi
-        if (continuations == null) {
-            return "";
+    private void collectMusicStreamsFrom(InfoItemsSearchCollector collector, JsonArray videos) {
+        final TimeAgoParser timeAgoParser = getTimeAgoParser();
+
+        for (Object item : videos) {
+            final JsonObject info = ((JsonObject) item).getObject("musicResponsiveListItemRenderer");
+            if (info != null) {
+                final String searchType = getLinkHandler().getContentFilters().get(0);
+                if (searchType.equals(MUSIC_SONGS) || searchType.equals(MUSIC_VIDEOS)) {
+                    collector.commit(new YoutubeStreamInfoItemExtractor(info, timeAgoParser) {
+                        @Override
+                        public String getUrl() throws ParsingException {
+                            String url = getUrlFromNavigationEndpoint(info.getObject("doubleTapCommand"));
+                            if (url != null && !url.isEmpty()) return url;
+                            throw new ParsingException("Could not get url");
+                        }
+
+                        @Override
+                        public String getName() throws ParsingException {
+                            String name = getTextFromObject(info.getArray("flexColumns").getObject(0)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (name != null && !name.isEmpty()) return name;
+                            throw new ParsingException("Could not get name");
+                        }
+
+                        @Override
+                        public long getDuration() throws ParsingException {
+                            String duration = getTextFromObject(info.getArray("flexColumns").getObject(3)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (duration != null && !duration.isEmpty())
+                                return YoutubeParsingHelper.parseDurationString(duration);
+                            throw new ParsingException("Could not get duration");
+                        }
+
+                        @Override
+                        public String getUploaderName() throws ParsingException {
+                            String name = getTextFromObject(info.getArray("flexColumns").getObject(1)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (name != null && !name.isEmpty()) return name;
+                            throw new ParsingException("Could not get uploader name");
+                        }
+
+                        @Override
+                        public String getUploaderUrl() throws ParsingException {
+                            if (searchType.equals(MUSIC_VIDEOS)) return null;
```

Comment:
It is available in some video items which have the "Go to artist" button in the item menu, though it is usually for the "topic" channel instead, maybe we should ignore them anyway?

We don't even use this uploader url for anything right now.

Reply:
I'll add support for them, though topic channels will only be useful once I made my follow-up tabs PR.